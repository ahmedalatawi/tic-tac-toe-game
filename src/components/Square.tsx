import { StyleSheet, Text, Dimensions, TouchableHighlight } from "react-native";
import { Colors } from "../constants/colors";

const { width, height } = Dimensions.get("window");

console.log(width, height);

interface Props {
  value: string | null;
  disabled?: boolean;
  testID?: string;
  onSquarePress: () => void;
}

function Square({ value, disabled, testID, onSquarePress }: Props) {
  return (
    <TouchableHighlight
      testID={testID}
      disabled={disabled}
      onPress={onSquarePress}
      style={styles.container}
      underlayColor={value ? "" : Colors.gray}
    >
      <Text style={styles.label}>{value}</Text>
    </TouchableHighlight>
  );
}

const styles = StyleSheet.create({
  container: {
    borderWidth: 1,
    flexDirection: "row",
    width: 76,
    height: 76,
    alignItems: "center",
    justifyContent: "center",
  },
  label: {
    fontSize: 40,
  },
});

export default Square;
