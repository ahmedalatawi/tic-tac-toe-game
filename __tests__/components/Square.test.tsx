import React from "react";
import renderer from "react-test-renderer";
import { render, screen, userEvent } from "@testing-library/react-native";
import Square from "../../src/components/Square";

const onSquarePressMock = jest.fn();
const propsMock = {
  value: "X",
  disabled: false,
  testID: "squareTestId",
  onSquarePress: onSquarePressMock,
};

describe("<Square />", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<Square {...propsMock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should display the correct value, e.g. X", async () => {
    render(<Square {...propsMock} />);

    const x = screen.getByText("X");

    expect(x).toBeDefined();
  });

  test("should fire onSquarePress when square is pressed", async () => {
    render(<Square {...propsMock} />);

    const x = screen.getByText("X");

    await userEvent.press(x);

    expect(onSquarePressMock).toHaveBeenCalled();
  });

  test("should fire not onSquarePress when square is disabled", async () => {
    const callBack = jest.fn();
    render(<Square {...propsMock} onSquarePress={callBack} disabled={true} />);

    const x = screen.getByText("X");

    await userEvent.press(x);

    expect(callBack).not.toHaveBeenCalled();
  });
});
