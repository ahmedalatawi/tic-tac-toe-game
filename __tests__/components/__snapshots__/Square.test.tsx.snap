// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "disabled": false,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 76,
      "justifyContent": "center",
      "width": 76,
    }
  }
  testID="squareTestId"
>
  <Text
    style={
      {
        "fontSize": 40,
      }
    }
  >
    X
  </Text>
</View>
`;
